### PR TITLE
Relationship viewer

### PR DESCRIPTION
**Pull request by MagikManz**
This plugin allows you to view your relationships within servers.

Fixed:
Crashing when un-able to load server profiles
Now profiles should load in with time if not immediately (due to ratelimits etc, it all depends on your amount of friends).
Slider no longer affects everything globally.

Video:

https://user-images.githubusercontent.com/36787474/232243291-518a3dd7-6836-4e14-a7c7-c6a5d58ad2d7.mp4

### CODE REVIEW

File: src/plugins/relationshipViewer/relationsStyles.css [hunk 1]
```diff
@@ -0,0 +1,80 @@
+:root {
+    --scrollbar-track: #ebebeb;
+    --scrollbar-thumb: #6d6d6d;
+    --expandable: #6d6d6d;
```
Comment (Vendicated) [inline src/plugins/relationshipViewer/relationsStyles.css:4]: avoid hardcoding colours. Always use Discord's css varialbles for proper theme support

File: src/plugins/relationshipViewer/relationsStyles.css [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+}
+
+.relation-form-container {
+    display: contents;
+    flex-direction: column;
+    gap: 0 !important;
+}
+
+.relation-closed {
+    overflow-y: hidden;
```
Comment (Vendicated) [inline src/plugins/relationshipViewer/relationsStyles.css:31]:
```suggestion
```

File: src/plugins/relationshipViewer/index.tsx [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+        document.documentElement.style.setProperty("--scrollbar-thumb", "#ffffff79");
+    } else {
+        document.documentElement.style.setProperty("--expandable", "#6d6d6d");
+        document.documentElement.style.setProperty("--scrollbar-track", "#ebebeb");
+        document.documentElement.style.setProperty("--scrollbar-thumb", "#6d6d6d");
+    }
+};
+
+const getUnloadedIds = (ids: Array<string>, guildId: string) => {
+    const unloadedIds: Array<string> = [];
```
Comment (Vendicated) [inline src/plugins/relationshipViewer/index.tsx:67]:
```suggestion
    return ids.filter(id => !GuildMemberStore.isMember(guildId, id));
```

**Review by Vendicated — CHANGES_REQUESTED** [review]

**Comment by MagikManz** [comment]
thnnx

**Comment by MagikManz** [comment]
🦗🦗🦗🦗🦗🦗🦗🦗🦗🦗🦗🦗🦗🦗🦗🦗🦗🦗🦗🦗
🦗🦗🦗🦗🦗🦗🦗🦗🦗🦗🦗🦗🦗🦗🦗🦗🦗🦗🦗🦗🦗🦗🦗🦗🦗🦗🦗🦗🦗🦗🦗🦗🦗🦗
(ven) 🦗🦗🦗🦗🦗🦗🦗🦗🦗🦗🦗🦗🦗🦗🦗🦗

**Comment by MagikManz** [comment]
🦗🦗🦗🦗🦗🦗🦗🦗🦗🦗🦗🦗🦗🦗

**Comment by nmsturcke** [comment]
BTW this has been added in the ServerProfile plugin (2c758ccdf8372c48f2d2a79ece8f2eba63f1510c)

**Comment by MagikManz** [comment]
yes I saw.... kinda disappointed tbh

**Pull request closed without merging**